redical_core/src/queries/event_instance_query.rs

```diff
@@ -83,6 +87,55 @@ impl<'cal> QueryIndexAccessor<'cal> for EventInstanceQueryIndexAccessor<'cal> {
             .unwrap_or(&InvertedCalendarIndexTerm::new())
             .to_owned()
     }
+
+    fn search_not_uid_index(&self, uid: &str) -> InvertedCalendarIndexTerm {
+        let mut inverse_matches = InvertedCalendarIndexTerm::new();
+        let mut event_uids = self.event_uids.clone();
+
+        event_uids.retain(|event_uid| event_uid != uid);
+
+        for event_uid in event_uids.into_iter() {
+            inverse_matches.insert_included_event(event_uid, None);
+        }
+
+        inverse_matches
```

Comment:
Could be faster to reference the `event_uids` instead of cloning and combine the filtering and `InvertedCalendarIndexTerm` building operations into one to save on memory movement and CPU cycles 🤔 

Something like:
```suggestion
        let mut inverse_matches = InvertedCalendarIndexTerm::new();

        for event_uid in self.event_uids.iter() {
            if event_uid == uid {
                continue;
            }

            inverse_matches.insert_included_event(event_uid.to_owned(), None);
        }

        inverse_matches
```

or

```suggestion
        let mut inverse_matches = InvertedCalendarIndexTerm::new();

        let include_event_uids_iter =
            self.event_uids
                .iter()
                .filter(|event_uid| *event_uid != uid);

        for event_uid in include_event_uids_iter {
            inverse_matches.insert_included_event(event_uid.to_owned(), None);
        }

        inverse_matches
```

Reply:
Great shout, will update!